### PR TITLE
disable telemetry with init

### LINKED ISSUE #364: `cyctl` telemetry flag
### Problem

When installing Cyclops with `kubectl` and the install manifests, you have the option to turn off telemetry. But when installing Cyclops with `cyctl` and the `init` command, this option does not exist. 

### Solution

Implement a `—-disable-telemetry` flag to disable telemetry when installing Cyclops via `cyctl`.

The command should look like the following:

```yaml
cyctl init —-disable-telemetry
```

### PR DESCRIPTION
**Pull request by thebigbone**
closes #364 

## 📑 Description
added a telemetry flag which will disable the telemetry

## ✅ Checks
- [ ] I have updated the documentation as required
- [x] I have performed a self-review of my code

### CODE REVIEW

File: cyctl/cmd/apply.go [hunk 1]
```diff
@@ -89,6 +89,44 @@ func applyYaml(yamlFile []byte, config *rest.Config) error {
 		if err != nil {
 			return err
 		}
+
+		if telemetry {
+			if obj.GetKind() == "Deployment" {
+				labels, _, _ := unstructured.NestedMap(obj.Object, "metadata", "labels")
+				if labels != nil {
+					appLabel, _, _ := unstructured.NestedString(labels, "app")
+					if appLabel == "cyclops-ctrl" {
```
Comment (petar-cvit) [inline cyctl/cmd/apply.go:98]: Can we check for cyclops controller deployment by checking the name? Also, you can check for all of those in the same if statement
```suggestion
		if telemetry && obj.GetKind() == "Deployment" && obj.GetName() == "cyclops-ctrl" {
```

File: cyctl/cmd/apply.go [hunk 2]
```diff
@@ -89,6 +89,44 @@ func applyYaml(yamlFile []byte, config *rest.Config) error {
...
+							if err != nil {
+								log.Fatal(err)
+							}
+							containers[0] = container
+							err = unstructured.SetNestedSlice(obj.Object, containers, "spec", "template", "spec", "containers")
+							if err != nil {
+								log.Fatal(err)
+							}
+						}
+						err = doServerSideApply(context.TODO(), config, obj)
```
Comment (petar-cvit) [inline cyctl/cmd/apply.go:118]: Are you first deploying the controller without telemetry and then updating it? I would prefer if just alter the object and then call the `doServerSideApply` with it

File: cyctl/cmd/apply.go [hunk 3]
```diff
@@ -161,5 +201,6 @@ var applyCmd = &cobra.Command{
 
 func init() {
 	applyCmd.Flags().StringP("version", "v", "main", "specify cyclops version")
+	applyCmd.Flags().BoolP("telemetry", "t", false, "disable telemetry")
```
Comment (petar-cvit) [inline cyctl/cmd/apply.go:204]: Can we rename the flag to `disable-telemetry` and update the usage string to something like `disable emitting telemetry metrics from cyclops controller`

**Review by petar-cvit — CHANGES_REQUESTED** [review]
Thanks @thebigbone, left a couple of comments

**Comment by thebigbone** [comment]
done

File: cyctl/cmd/apply.go [hunk 4]
```diff
@@ -103,6 +126,8 @@ var applyCmd = &cobra.Command{
 			log.Fatal(err)
 		}
 
+		telemetry, _ := cmd.Flags().GetBool("disable-telemetry")
```
Comment (petar-cvit) [inline cyctl/cmd/apply.go:129]: Can you handle the error and rename the variable to `disableTelemetry`?
```suggestion
		disableTelemetry, err := cmd.Flags().GetBool("disable-telemetry")
		if err != nil {
			log.Fatal(err)
		}
```

**Review by petar-cvit — CHANGES_REQUESTED** [review]

**Comment by thebigbone** [comment]
done

File: cyctl/cmd/apply.go [hunk 5]
```diff
@@ -67,7 +67,7 @@ func doServerSideApply(ctx context.Context, cfg *rest.Config, obj *unstructured.
 	return err
 }
 
-func applyYaml(yamlFile []byte, config *rest.Config) error {
+func applyYaml(yamlFile []byte, config *rest.Config, telemetry bool) error {
```
Comment (petar-cvit) [inline cyctl/cmd/apply.go:70]: Can you also rename this one?

**Review by petar-cvit — CHANGES_REQUESTED** [review]

File: cyctl/cmd/apply.go (hunk 5 above)
Comment (thebigbone) [reply cyctl/cmd/apply.go:70]: done

**Review by petar-cvit — APPROVED** [review]
Thanks @thebigbone 🧡

**Pull request merged**